### PR TITLE
Script for reproducing issue 965

### PR DESCRIPTION
**Pull request by HSavynetska**
Script for reproducing issue [[API] App is not unregistered by reason = REQUEST_WHILE_IN_NONE_HMI_LEVEL](https://github.com/smartdevicelink/sdl_core/issues/965)

### CODE REVIEW

**Comment by HSavynetska** [comment]
The defect is not reproduced
sdl_core: branch develop
commit: 65c1e7e
Verified with script:
Script is under review
https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/1902

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+			return true
+		elseif
+			data.payload.resultCode == "APPLICATION_NOT_REGISTERED" then
+			AppNotRegistered = AppNotRegistered +1
+			return true
+		end
+	end)
+	:Times(AtMost(10))
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+	self.mobileSession1:ExpectNotification("OnAppInterfaceUnregistered", {{reason = "REQUEST_WHILE_IN_NONE_HMI_LEVEL"}})
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: parameters are defined as array, please define as in MOBILE_API

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Update .ini file", updateINIFile)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("RAI, PTU", common.rai_ptu_n)
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: can be used rai_n function without ptu

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+--[[ Local Functions ]]
+local function updateINIFile()
+  common.backupINIFile()
+  commonFunctions:write_parameter_to_smart_device_link_ini("AppHMILevelNoneTimeScaleMaxRequests", count_of_requests)
+  commonFunctions:write_parameter_to_smart_device_link_ini("AppHMILevelNoneRequestsTimeScale", 30000)
+end
+
+local function listFilesRequests(self)
+	for i = 1, 10 do
+	self.mobileSession1:SendRPC("ListFiles", {})
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: please update with align

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	end
+	self.mobileSession1:ExpectResponse("ListFiles")
+	:Do(function(_, data)
+		if
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: please update to one line if with condition

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+			data.payload.resultCode == "SUCCESS" then
+			return true
+		elseif
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: please update to one line if with condition

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+			data.payload.resultCode == "APPLICATION_NOT_REGISTERED" then
+			return true
+		else return false, "Received unexpected resultCode " .. data.payload.resultCode
+		end
+	end)
+	:Times(AtMost(10))
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: Why Times(AtMost(10))? Can SDL receive less then 10 responses?

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", {appID = common.getHMIAppId(pAppId), unexpectedDisconnect = false})
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: pAppId is empty, please update

File: test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	self.mobileSession1:ExpectNotification("OnAppInterfaceUnregistered", {{reason = reason }})
```
Comment (GetmanetsIrina) [inline test_scripts/Defects/4_6/965_API_App_is_not_unregistered_by_reason_=_REQUEST_WHILE_IN_NONE_HMI_LEVEL.lua]: reason value is empty please update

**Pull request merged**